lib/Doctrine/ORM/Cache/DefaultCacheFactory.php

```diff
@@ -47,9 +49,33 @@ class DefaultCacheFactory implements CacheFactory
     /** @var string|null */
     private $fileLockRegionDirectory;
 
-    public function __construct(RegionsConfiguration $cacheConfig, CacheAdapter $cache)
+    /**
+     * @param CacheItemPoolInterface $cacheItemPool
```

Comment:
I'd also document the legacy types, so as not to confuse static analysis

Reply:
I intentionally documented the non-deprecated usage only. If downstream projects used a static analysis tool, that tool would start to emit errors which is a good thing imho. It makes this kind of deprecation discoverable.

Reply:
I think we should stay away from it for the reason @orklah mentioned here: https://symfony-devs.slack.com/archives/C8SFXTD2M/p1641293007273300

Reply:
That discussion did not really convince me, I must admit.

Reply:
Let's not block this PR for that then.